server/routes/api/users.js

```diff
@@ -75,4 +76,78 @@ router.post("/login", (req, res) => {
   }
 });
 
+// @route GET /api/users/{id}
+// @desc Get a User object
+// @access Authenticated User can get own record
+router.get("/:id", (req, res) => {
+  let id = req.params.id;
+  User.findById(id, function(err, user) {
+    if (!user) {
+      res.status(404).send({ id: `User with id ${id} is not found` });
+    } else {
+      res.json(user);
+    }
+  });
+});
+
+// @route PUT /api/users/{id}
+// @desc Update a User object
+// @access Authenticated Users
+router.put("/:id", (req, res) => {
+  let id = req.params.id;
+
+  // TO DO *****************
+  const { errors, isValid } = validateUserInput(req.body);
+
+  if (!isValid) {
+    return res.status(400).json(errors);
+  }
+
+  User.findById(id, function(err, user) {
+    if (!user) {
+      res.status(404).send({ id: `User with id ${id} is not found` });
+    } else {
+      //if updating email field, check if User with new email alredy exists
+      if (req.body.email !== user.email) {
+        User.findOne({ email: req.body.email }).then(userEmailCheck => {
+          if (userEmailCheck) {
+            return res.status(400).json({
+              email:
+                "Cannot update email address. Email address already exists",
+            });
+          }
+        });
+      }
+      user.firstName = req.body.firstName ? req.body.firstName : user.firstName;
+      user.lastName = req.body.lastName ? req.body.lastName : user.lastName;
+      user.email = req.body.email ? req.body.email : user.email;
+
+      user
+        .save()
+        .then(user => {
+          res.json(user);
+        })
+        .catch(err => console.log(err));
+    }
+  });
+});
+
+// @route DELETE /api/users/{id}
+// @desc Delete a User object
+// @access Authenticated Users
+router.delete("/:id", (req, res) => {
```

Comment:
Don't know if we need this route accessibly by the user - probably would be an admin route

Reply:
I've removed this for now as we have no admin user role.